src/strategy.ts

```diff
@@ -0,0 +1,184 @@
+import { Strategy as OAuth2Strategy } from 'passport-oauth2';
+
+import { mapUserProfile } from './mapUserProfile';
+import { Profile, ProfileWithMetaData } from './models/profile';
+import { StrategyOption } from './models/strategyOptions';
+import { TwitterError } from './models/twitterError';
+import { TwitterUserInfoResponse } from './models/twitterUserInfo';
+
+export class Strategy extends OAuth2Strategy {
+  _userProfileURL: string;
+
+  /**
+   * Twitter strategy constructor
+   *
+   * The Twitter authentication strategy authenticates requests by delegating to
+   * Twitter using the OAuth 2.0 protocol.
+   *
+   * Applications must supply a `verify` callback which accepts an `accessToken`,
+   * `refreshToken` and service-specific `profile`, and then calls the `cb`
+   * callback supplying a `user`, which should be set to `false` if the
+   * credentials are not valid.  If an exception occured, `err` should be set.
+   *
+   * Required options:
+   *   - `clientID`      your Twitter application's App ID
+   *   - `clientSecret`  your Twitter application's App Secret
+   *   - `callbackURL`   URL to which Twitter will redirect the user after granting authorization
+   *   - `clientType`    your Twitter application (client) type, either `public` or `confidental`
+   *
+   * Examples:
+   *
+   *     passport.use(new TwitterStrategy({
+   *         clientType: 'confidential',
+   *         clientID: 'client-identification',
+   *         clientSecret: 'secret'
+   *         callbackURL: 'https://www.example.net/auth/twitter/callback'
+   *       },
+   *       function(accessToken, refreshToken, profile, cb) {
+   *         User.findOrCreate(..., function (err, user) {
+   *           cb(err, user);
+   *         });
+   *       }
+   *     ));
+   */
```

Comment:
suggestion (docs): Consider using JSDoc / [TSDoc](https://tsdoc.org/) tags in the doc string, as most IDEs generate nicer documentation from that. For example:

```suggestion
  /**
   * Twitter strategy constructor
   *
   * Required options:
   *
   *   - `clientID` - your Twitter application's App ID
   *   - `clientSecret` - your Twitter application's App Secret
   *   - `callbackURL` - URL to which Twitter will redirect the user after granting authorization
   *   - `clientType` - your Twitter application (client) type, either `public` or `confidental`
   *
   * @remarks
   * The Twitter authentication strategy authenticates requests by delegating to
   * Twitter using the OAuth 2.0 protocol.
   *
   * Applications must supply a `verify` callback which accepts an `accessToken`,
   * `refreshToken` and service-specific `profile`, and then calls the `cb`
   * callback supplying a `user`, which should be set to `false` if the
   * credentials are not valid.  If an exception occured, `err` should be set.
   *
   * @example
   * passport.use(new TwitterStrategy({
   *     clientType: 'confidential',
   *     clientID: 'client-identification',
   *     clientSecret: 'secret'
   *     callbackURL: 'https://www.example.net/auth/twitter/callback'
   *   },
   *   function(accessToken, refreshToken, profile, cb) {
   *     User.findOrCreate(..., function (err, user) {
   *       cb(err, user);
   *     });
   *   }
   * ));
   */
```

Reply:
Doc strings rewritten in TsDoc format. Thanks.